src/lib/state.ts

```diff
@@ -21,11 +21,57 @@ export { assertStatePrecondition, cleanStatePrecondition };
  * Gettable and settable state that can be checked for equality.
  */
 type State<A> = {
+  /**
```

Comment:
Calling for the help of @barriebyron here for these doc comments :smile: Would you happen to have any suggestions for the wording of these doc comments? Just for context, these comments will be shown to developers when they inspect the methods and in our API Reference docs.